From f0bfa1da54eee9614224c9a884855ff77beb79cd Mon Sep 17 00:00:00 2001
From: Eric Yang <Eric.Yang2@amd.com>
Date: Tue, 3 Oct 2017 18:15:51 -0400
Subject: [PATCH 4098/5855] drm/amd/display: Handle case when stream not found
 in set_dpms

When validate with context fail to add stream to the context, we have
a case where set_dpms won't be able to find the stream that need to
be disabled.

Signed-off-by: Eric Yang <Eric.Yang2@amd.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc.c b/drivers/gpu/drm/amd/display/dc/core/dc.c
index e257427..2e1245c 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc.c
@@ -363,7 +363,7 @@ void set_dpms(
 	struct dc_stream_state *stream,
 	bool dpms_off)
 {
-	struct pipe_ctx *pipe_ctx;
+	struct pipe_ctx *pipe_ctx = NULL;
 	int i;
 
 	for (i = 0; i < MAX_PIPES; i++) {
@@ -373,6 +373,11 @@ void set_dpms(
 		}
 	}
 
+	if (!pipe_ctx) {
+		ASSERT(0);
+		return;
+	}
+
 	if (stream->dpms_off != dpms_off) {
 		stream->dpms_off = dpms_off;
 		if (dpms_off)
-- 
2.7.4

